grpc/src/main/java/com/linecorp/armeria/server/grpc/FramedGrpcService.java

```diff
@@ -164,21 +168,24 @@ protected HttpResponse doPost(ServiceRequestContext ctx, HttpRequest req) throws
 
         ctx.logBuilder().serializationFormat(serializationFormat);
 
-        final String methodName = GrpcRequestUtil.determineMethod(ctx);
-        if (methodName == null) {
-            return HttpResponse.of(HttpStatus.BAD_REQUEST,
-                                   MediaType.PLAIN_TEXT_UTF_8,
-                                   "Invalid path.");
-        }
-
-        final ServerMethodDefinition<?, ?> method = registry.lookupMethod(methodName);
+        ServerMethodDefinition<?, ?> method = ctx.attr(RESOLVED_GRPC_METHOD);
```

Comment:
How about injecting `enableUnframedRequests` when creating `FramedGrpcService` and check the boolean condition before the hash lookup?

Reply:
Sorry, I don't understand what you want. Just set `enableUnframedRequests` as `true` when configuring `UnframedGrpcService` or `HttpJsonTranscodingService` and then use the flag like below?

```
ServerMethodDefinition<?, ?> method = enableUnframedRequests ? registry.lookupMethod(methodName) : null;
```